lib/vagrant/ui.rb

```diff
@@ -31,9 +31,14 @@ class Interface
       attr_accessor :stderr
 
       def initialize
-        @logger   = Log4r::Logger.new("vagrant::ui::interface")
-        @opts     = {}
+        if ENV["VAGRANT_QUIET_PROGRESS"]
+          show_progress = false
+        else
+          show_progress = true
+        end
 
+        @logger   = Log4r::Logger.new("vagrant::ui::interface")
+        @opts     = { :show_progress => show_progress }
```

Comment:
`show_progress` options can be ignored since the subclass usage will implicitly provide the logic